src/gui/ScreenHandler.cpp

```diff
@@ -146,6 +146,16 @@ void Screens::CloseAll() {
     close_all = true;
 }
 
+void Screens::CloseSerial() {
+    /// serial close logic:
+    /// when serial printing screen (M876) is open, Screens::SerialClose() is
+    /// called and it will iterate all screens to close those that should be closed
+    while (current.get()->IsSerial() || stack_iterator != stack.begin()) {
```

Comment:
Are you absolutely sure the current always returns a valid pointer in this case?
The same applies to the stack_iterator and stack.begin()
... not saying it is wrong, just asking...

Reply:
I think it is safe - now. But if something changes it could cause bsod.
So pls add check.